server/db/dbcontrollers/matchesController.js

```diff
@@ -0,0 +1,69 @@
+//matched controllers WORK OUT FOR HEAD TO HEADS!!!
+module.exports = function (knex) {
+  var module = {};
+
+  //how to get multiple ids when have head to head matches--------//
+  /*knex.select('name').from('users')
+    .whereIn('id', [1, 2, 3])
+    .orWhereIn('id', [4, 5, 6])*/
+  //bit of an over kill dont at this point need to search user ID 
+  //from table at this point! Left in as not hurting us to be tidied
+  //-------------------create a match-----------------------------//
+
+  module.createMatch = function (userID, startFunds, type) {
+
+
+    //TODO: match logic
+    var nextWeek = Date(Date.now() + (7 * 24 * 60 * 60 * 1000));
+    var startDate = new Date();
+    var endDate = nextWeek;
+    var status = 'in progress';
+    var title = 'solo match';
+
+    return knex('matches').insert({
+        'title': title,
+        'creator_id': userID,
+        'starting_funds': startFunds,
+        'challengee': userID,
+        'startdate': startDate,
+        'enddate': endDate,
+        'status': status,
+        'type': type
+      })
+      //echo back succesful insert
+      .then(function () {
+        return knex.select()
+          .table('matches')
+          .where({
+            'title': title,
+            'creator_id': userID,
+            'starting_funds': startFunds,
+            'challengee': userID,
+            'startdate': startDate,
+            'enddate': endDate,
+            'status': status,
+            'type': type
+          });
+      })
+      .then(function (match) {
+        return match[0];
+      });
+
```

Comment:
Question from anna

1) // Returns [2] in "mysql", "sqlite"; [2, 3] in "postgresql"
knex('books')
  .returning('id')
  .insert([{title: 'Great Gatsby'}, {title: 'Fahrenheit 451'}])
2)knex('table').insert({a: 'b'}).returning('*').toString();